plugins/BEdita/Core/tests/TestCase/Model/Entity/StreamTest.php

```diff
@@ -327,4 +329,72 @@ public function testGetUrl($expected, $uuid)
         static::assertSame($expected, $first);
         static::assertSame($expected, $second);
     }
+
+    /**
+     * Read data from image if is possible
+     * @param Stream $stream stream entity
+     */
+
+    public function readDataFromImage($stream): void
+    {
+        if (!preg_match('/image\//', $stream->mime_type)) {
+            static::assertNull($stream->width);
+            static::assertNull($stream->height);
+        } else {
+            if (function_exists('getimagesizefromstring')) {
+                static::assertNotNull($stream->width);
+                static::assertNotNull($stream->height);
+            } else {
+                static::assertNull($stream->width);
+                static::assertNull($stream->height);
+            }
+
+            if (function_exists('exif_read_data') && in_array($stream->mime_type, $stream::EXIF_MIME_TYPES)) {
+                static::assertNotNull($stream->file_metadata);
+            } else {
+                static::assertEmpty($stream->file_metadata);
+            }
+        }
+    }
+
+    /**
+     * Test read exif data
+     *
+     * @covers ::readFileMetadata()
+     * @covers ::createStream()
+     */
+    public function testReadFileMetadata()
+    {
+        $path = Configure::read('Filesystem.default.path');
+        $image_test = new Stream($path . '/a4fbe302-3d5b-4774-a9df-18598def690e-image-metadata.jpeg', 'r');
+        $gif_test = new Stream($path . '/6aceb0eb-bd30-4f60-ac74-273083b921b6-bedita-logo-gray.gif', 'r');
```

Comment:
[minor] for convention we use camel case for variables `$imageTest`, `$gifTest`, etc...